src/vm-repair/azext_vm_repair/_validators.py

```diff
@@ -287,26 +288,18 @@ def _validate_disk_name(disk_name):
 
 
 def _validate_resource_group_name(rg_name):
+    from knack.prompting import prompt_y_n
     rg_pattern = r'[0-9a-zA-Z._\-()]+$'
     # if match is null or ends in period, then raise error
     if not match(rg_pattern, rg_name) or rg_name[-1] == '.':
         raise CLIError('Resource group name only allow alphanumeric characters, periods, underscores, hyphens and parenthesis and cannot end in a period.')
 
     if len(rg_name) > 90:
         raise CLIError('Resource group name only allow up to 90 characters.')
-
-    # Check for existing dup name
-    try:
-        list_rg_command = 'az group list --query "[].name" -o json'
-        logger.info('Checking for existing resource groups with identical name within subscription...')
-        output = _call_az_command(list_rg_command)
-    except AzCommandError as azCommandError:
-        logger.error(azCommandError)
-        raise CLIError('Unexpected error occured while fetching existing resource groups.')
-    rg_list = loads(output)
-
-    if rg_name in [rg.lower() for rg in rg_list]:
-        raise CLIError('Resource group with name \'{}\' already exists within subscription.'.format(rg_name))
+    if _check_existing_rg(rg_name):
+        if not prompt_y_n('Resource Group already exists. Continue to use existing resource group? If operation fails you will prompted to delete resource group'):
+            raise CLIError('Resource group with name \'{}\' already exists within subscription.'.format(rg_name))
```

Comment:
May I ask does the new step of prompt may block the user's automation script?

Reply:
Vm-repair is not used in automation because it is used for manual intervention when vm is unbootable. And when an existing rg is being used the customer needs to be warned that their data in existing resource group will be deleted.

Reply:
OK, got it~ Thanks